azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/inbox_manager.py

```diff
@@ -0,0 +1,81 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a manager for inboxes."""
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class InboxManager(object):
+    """Manages the various Inboxes for a client.
+
+    :ivar c2d_message_inbox: The C2D message Inbox.
+    :ivar input_message_inboxes: A dictionary mapping input names to input message Inboxes.
+    """
+
+    def __init__(self, inbox_type):
+        """Initializer for the InboxManager.
+
+        :param enable_feature_fn: A function that can enable transport features.
+        :param inbox_type: An Inbox class that the manager will use to create Inboxes.
+        """
+        self._create_inbox = inbox_type
```

Comment:
>self._create_inbox = inbox_type [](start = 8, length = 31)

nit: sematics.  inbox_type has some meaning, but I would expect something with that name to be a string.  _create_inbox also has meaning, but feels awkward.  maybe inbox_initializer?
(if the parameter name doesn't match the attribute, that's a smell -- I always feel awkward when I have 2 different names and it usually means I haven't decided what that thing is exactly.)

Reply:
The reasoning as is, is that you pass the literal type through as inbox_type. SyncClientInbox for instance. This gets put into the _create_inbox variable, because, a type is callable (e.g. SyncClientInbox())

The context is different from when the thing is supplied, to when it's used, but it is still very much the same thing.

I have some ideas that can make this pattern of agnostic class that operates on specifically typed object better and clearer in this exact domain (passing through the object type for it to work open), but they're hard to justify at the moment given that this is currently the only class using it, and it would create additional infrastructure.

---
In reply to: [264336946](https://github.com/Azure/azure-iot-sdk-python-preview/pull/55#discussion_r264336946) [](ancestors = 264336946)